src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -29330,14 +29323,43 @@ static void M1<T>(T t1)
                 Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "t5").WithLocation(35, 30),
                 // (36,31): warning CS8607: A possible null value may not be assigned to a target marked with the [DisallowNull] attribute
                 //         new CStruct<T>().P5 = t5; // 9
-                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "t5").WithLocation(36, 31)
-            };
-
-            var comp = CreateNullableCompilation(source, references: new[] { lib.EmitToImageReference() });
-            comp.VerifyDiagnostics(expected);
+                Diagnostic(ErrorCode.WRN_DisallowNullAttributeForbidsMaybeNullAssignment, "t5").WithLocation(36, 31));
 
             var comp2 = CreateNullableCompilation(new[] { source, lib_cs, DisallowNullAttributeDefinition });
```

Comment:
Do the diagnostics now vary based on whether the base is from source or metadata?

Reply:
Yes and no. The diagnostics for `comp` include `source` only whereas the diagnostics for `comp2` include both `source` and `lib_cs`.

---
In reply to: [349356225](https://github.com/dotnet/roslyn/pull/39939#discussion_r349356225) [](ancestors = 349356225)

Reply:
Thanks, it's much more obvious now that I look at the actual file, GitHub was just hiding bits from me.

---
In reply to: [349358281](https://github.com/dotnet/roslyn/pull/39939#discussion_r349358281) [](ancestors = 349358281,349356225)